### PR TITLE
Convert keys for unmanaged source and javadoc archives to tasks

### LINKED ISSUE #353: ensimeUnmanagedSourceArchives should be a Task
Everything should be a `Task` really because it stinks if we need to depend on something that is a `Task`. 

e.g. legit usecase for intellij-scala

```scala
ensimeUnmanagedSourceArchives in LocalProject("scalaCommunity") := Seq(ideaBaseDirectory.value / "sources.zip")
```

but RHS is a Task...

### PR DESCRIPTION
**Pull request by yyadavalli**
Resolve #353

### CODE REVIEW

File: src/main/scala/EnsimePlugin.scala [hunk 1]
```diff
@@ -427,12 +427,12 @@ object EnsimePlugin extends AutoPlugin {
     def jarSrcsFor(config: Configuration) = updateClassifiersReport.select(
       configuration = configFilter(config),
       artifact = artifactFilter(classifier = Artifact.SourceClassifier)
-    ).toSet ++ (ensimeUnmanagedSourceArchives in config in projectRef).gimme
```
Comment (fommil) [inline src/main/scala/EnsimePlugin.scala:430]: hmm, I thought we got rid of the command support stuff when you converted us to Task, how come `gimme` was still around?

Comment (yyadavalli) [reply src/main/scala/EnsimePlugin.scala:430]: `gimme` was still used in the tasks, I remember its about using a dynamic configuration when getting a setting value.

Comment (fommil) [reply src/main/scala/EnsimePlugin.scala:430]: I think it was always supposed to be a hack to access settings / tasks from commands :-D it appears to have been repurposed

Comment (yyadavalli) [reply src/main/scala/EnsimePlugin.scala:430]: Yeah, in `projectData` methods like `sourcesFor` cannot accept a `Configuration` as a parameter if we were to use the task macro. So I might have reused `gimme`.

**Comment by fommil** [comment]
sorry, was waiting for CI and it was running slow and I forgot.

**Pull request merged**